OpenRA.Mods.Common/Scripting/Global/TriggerGlobal.cs

```diff
@@ -470,6 +470,19 @@ public void OnSold(Actor a, LuaFunction func)
 			GetScriptTriggers(a).RegisterCallback(Trigger.OnSold, func, Context);
 		}
 
+		[Desc("Call a function when a time limit warning is hit. The callback function will be called as " +
+			"func(int minutesRemaining, string audioNotification).")]
```

Comment:
All our other times are in ticks, so using minutes here doesn't feel right.

Reply:
It's in minutes because the index for the `TimeLimitWarnings` dictionary is also in minutes, i.e. that's the lowest resolution you get. Do you prefer that to be changed as well? I can do that, but I'd argue that that would make the YAML config pretty much incomprehensible.

Reply:
I'm not sure what I prefer here. My motivation was that after all the arguments and effort that it took to standardize timings around ticks, seeing new "realtime" times and the walltime parameter triggers mild PTSD.

Reply:
Another thought to consider is whether having the time resolution at minutes counts as a "cncism", since this is built around the audio notifications shipped in the original games.

Reply:
Looking at the code in more detail: why do we need to index the notifications / print to the chat from lua at all? The `TimeLimitManager` can handle this for us.

This then provides a nice rug to sweep the timestep/game speed differences under: times in Lua remain always in ticks, and the trait code handles converting this to a wallclock time that is reported to the player.
We'd want to keep the `OnTimerExpired` trigger, but any gameplay-affecting intermediate time triggers would be done using the standard `AfterDelay` trigger.

Reply:
> using a custom prefix (e.g. "Convoy approaching in")

That would solve the issue, yes. At the moment many of the things you pointed out ("walltime", ignoring the replay timestep etc.) are meant to keep the time as printed from Lua scripts and the time shown on the game timer in sync. I think it would be awkward for one of them to show half or double the time.  But you suggestions could be a reasonable solution to that, I'll have a go at that as soon as I get a chance.

Reply:
Good point about the timer banner. IMO it would be good to standardize this between SP and MP, and have it shown somewhere more prominent than the game timer but perhaps less prominent than the centered banner in both places.